src/nacl/bindings/crypto_generichash.py

```diff
@@ -159,58 +179,57 @@ def generichash_blake2b_init(key=b'', salt=b'',
     ffi.memmove(_salt, salt, len(salt))
     ffi.memmove(_person, person, len(person))
 
-    rc = lib.crypto_generichash_blake2b_init_salt_personal(statebuf,
+    rc = lib.crypto_generichash_blake2b_init_salt_personal(state.statebuf,
                                                            key, len(key),
                                                            digest_size,
                                                            _salt, _person)
     ensure(rc == 0, 'Unexpected failure',
            raising=exc.RuntimeError)
 
-    return statebuf
+    return state
 
 
-def generichash_blake2b_update(statebuf, data):
+def generichash_blake2b_update(state, data):
     """Update the blake2b hash state
 
-    :param statebuf: an initialized blake2b state buffer as returned from
+    :param state: a initialized Blake2bState object as returned from
                      :py:func:`.crypto_generichash_blake2b_init`
-    :type statebuf: object
+    :type state: :py:class:`.Blake2State`
     :param data:
     :type data: bytes
     """
 
+    ensure(isinstance(state, Blake2State),
+           'State must be a Blake2State object',
+           raising=exc.TypeError)
+
     ensure(isinstance(data, bytes),
            'Input data must be a bytes sequence',
            raising=exc.TypeError)
 
-    rc = lib.crypto_generichash_blake2b_update(statebuf, data, len(data))
+    rc = lib.crypto_generichash_blake2b_update(state.statebuf, data, len(data))
     ensure(rc == 0, 'Unexpected failure',
            raising=exc.RuntimeError)
 
 
-def generichash_blake2b_final(statebuf, digest_size):
+def generichash_blake2b_final(state):
```

Comment:
Do we consider these bindings public API? If so this is a compatiblity break.

Reply:
Since our public documentation doesn't mention low level bindings, I think we can consider anything nacl.bindings a private API.

Reply:
At least unti #494 gets in, or someone starts writing explicit docs for low-level bindings :wink: